drakrun/drakrun/main.py

```diff
@@ -180,6 +182,29 @@ def default_low_timeout(self) -> int:
             "drakrun", "analysis_low_timeout", fallback=self.default_timeout
         )
 
+    def timeout_for_task(self, task: Task) -> int:
+        """Return a timeout for task - a default for quality or specified by task."""
+        default_timeout = (
+            self.default_low_timeout
+            if task.headers.get("quality", "high") == "low"
+            else self.default_timeout
+        )
+        return task.payload.get("timeout", default_timeout)
+
+    def filename_for_task(self, task: Task, magic_output: str) -> Tuple[str, str]:
+        """Return a tuple of (filename, extension) for a given task.
+        This depends on the content magic, "extension" header and "file_name" payload.
+        """
+        extension = self._karton_safe_get_headers(task, "extension", "exe").lower()
+        if "(DLL)" in magic_output:
+            extension = "dll"
```

Comment:
Is it good time to get rid of that _karton_safe_get_headers call and do:
```suggestion
        # headers["extensions"] may exist and be None
        extension = (task.headers.get("extension") or "exe").lower()
        if "(DLL)" in magic_output:
            extension = "dll"
```